File: src/main/java/com/github/catvod/crawler/Spider.java
package com.github.catvod.crawler;

import android.content.Context;
import okhttp3.Dns;
import org.json.JSONObject;

import java.util.HashMap;
import java.util.List;

public abstract class Spider {

    public static final JSONObject empty = new JSONObject();

    protected static Context mContext;
    protected static SpiderApi mSpiderApi;

    public void init(Context context) {
        mContext = context;
    }

    public void init(Context context, String extend) {
        init(context);
    }

    public String homeContent(boolean filter) {
        return "";
    }

    public String homeVideoContent() {
        return "";
    }

    public String categoryContent(String tid, String pg, boolean filter, HashMap<String, String> extend) {
        return "";
    }

    public String detailContent(List<String> ids) {
        return "";
    }

    public String searchContent(String key, boolean quick) {
        return "";
    }

    public String playerContent(String flag, String id, List<String> vipFlags) {
        return "";
    }

    public void initApi(SpiderApi spiderApi) {
        mSpiderApi = spiderApi;
    }

    public boolean isVideoFormat(String url) {
        return false;
    }

    public boolean manualVideoCheck() {
        return false;
    }

    public static Dns safeDns() {
        return Dns.SYSTEM;
    }
}

File: scripts/python_runtime/base/spider.py
from base64 import b64decode, b64encode

import requests


requests.packages.urllib3.disable_warnings()


class Spider:
    proxyUrl = "http://127.0.0.1:18080/proxy"

    def init(self, extend=""):
        return None

    def getName(self):
        return ""

    def isVideoFormat(self, url):
        return False

    def manualVideoCheck(self):
        return False

    def homeContent(self, filter):
        return {}

    def homeVideoContent(self):
        return {}

    def categoryContent(self, tid, pg, filter, extend):
        return {}

    def detailContent(self, ids):
        return {}

    def searchContent(self, key, quick, pg="1"):
        return {}

    def playerContent(self, flag, id, vipFlags=None):
        return {}

    def fetch(self, url, method="GET", headers=None, params=None, data=None, json=None, cookies=None,
              allow_redirects=True, timeout=10, **kwargs):
        request_method = (method or "GET").upper()
        return requests.request(
            request_method,
            url,
            headers=headers,
            params=params,
            data=data,
            json=json,
            cookies=cookies,
            allow_redirects=allow_redirects,
            timeout=timeout,
            verify=False,
            **kwargs,
        )

    def post(self, url, data=None, json=None, headers=None, timeout=10, **kwargs):
        return self.fetch(url, method="POST", data=data, json=json, headers=headers, timeout=timeout, **kwargs)

    def getProxyUrl(self):
        return getattr(self, "proxyUrl", self.proxyUrl)

    def e64(self, text):
        return b64encode((text or "").encode("utf-8")).decode("utf-8")

    def d64(self, text):
        return b64decode((text or "").encode("utf-8")).decode("utf-8")
